src/routes/transactions/routes.rs

```diff
@@ -66,21 +66,22 @@ pub async fn post_confirmation<'e>(
     safe_tx_hash: String,
     tx_confirmation_request: Result<Json<ConfirmationRequest>, Error<'e>>,
 ) -> ApiResult<content::Json<String>> {
-    CacheResponse::new(&context)
-        .resp_generator(|| {
-            proposal::submit_confirmation(
-                &context,
-                &chain_id,
-                &safe_tx_hash,
-                &tx_confirmation_request
-                    .as_ref()
-                    .unwrap()
-                    .0
-                    .signed_safe_tx_hash,
-            )
-        })
+    let request: ConfirmationRequest = tx_confirmation_request?.0;
+
+    proposal::submit_confirmation(
+        &context,
+        &chain_id,
+        &safe_tx_hash,
+        &request.signed_safe_tx_hash,
+    )
+    .await?;
+
+    let tx_details = CacheResponse::new(&context)
+        .resp_generator(|| details::get_transactions_details(&context, &chain_id, &safe_tx_hash))
```

Comment:
Do we want to cache the transaction details here?

Can't this `POST` affect the `TransactionStatus`?

Reply:
It does. We invalidate the cache right after the post request and only then we are able to reach this statement where we return the tx details